src/server/models/mediaquery.js

```diff
@@ -43,6 +43,19 @@ module.exports.getVideoResCountByCourse = async courseLabel => {
   return arrayOfVideoRes.length;
 };
 
+module.exports.getMusicResByCourse = async courseLabel => {
+  const musicResCollection = client.db(DB_DATABASE).collection('musicreserves');
+  const toBeReturned = await musicResCollection
+    .find({ classShortname: courseLabel })
+    .toArray();
+  return toBeReturned;
+};
+
+module.exports.getMusicResCountByCourse = async courseLabel => {
+  const arrayOfMusicRes = await this.getMusicResByCourse(courseLabel);
```

Comment:
@maxpluez For all these end points, you should be ensuring the user has a role and belongs to that given label. Or get the label from the ContextToken.

Reply:
Added role check under api route /api/media/index.js, which calls this function.